src/core/teleop_session_manager/python/input_selector.py

```
@@ -0,0 +1,170 @@
+    - ``float`` for analog signals thresholded into bool.
+
+    For float signals, hysteresis is applied:
+    - transition to True when value >= ``threshold``
+    - transition to False when value < ``release_threshold``
+    If ``release_threshold`` is omitted, it is set to ``threshold``.
+    """
+    if not isinstance(source_output_selector, OutputSelector):
+        raise TypeError(
+            "source_output_selector must be OutputSelector, "
+            f"got {type(source_output_selector).__name__}"
+        )
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)